lib/services/notification-api-service.js

```diff
@@ -19,45 +20,46 @@ di.annotate(notificationApiServiceFactory,
     )
 );
 
-function notificationApiServiceFactory(
+function NotificationApiServiceFactory(
     eventsProtocol,
     taskProtocol,
     taskGraphStore,
+    TaskGraph, 
     Logger,
     waterline,
     Errors,
     Promise,
     _
 ) {
-    var logger = Logger.initialize(notificationApiServiceFactory);
+    var logger = Logger.initialize(NotificationApiServiceFactory);
 
-    function notificationApiService() {
+    function NotificationApiService() {
     }
 
-    notificationApiService.prototype.postNotification = function(message) {
+    NotificationApiService.prototype.postNotification = function(message) {
         var self = this;
 
         if (_.has(message, 'nodeId')) {
             return self.postNodeNotification(message);
+        } else if (_.has(message, 'taskId') && _.has(message, 'progress')) {
+            // This will be progress update notification if taskId is specified
+            return self.postProgressEvent(message);
```

Comment:
I would suggest to refactor this to following so other develops can choose to develop other task notifications other than progress notification. 

```
       } else if (_.has(message, 'taskId')) {
            if (_.has(message, 'progress'))
            {
                // This will be progress update notification if taskId is specified
                return self.postProgressEvent(message);
            }
            if (_.has(message, 'SOMETHING_ELSE'))
            {
                // others can choose to post other kind of task notification without need to modify your code. 
            }
```